lib/chef/chef_fs/file_system/repository/chef_repository_file_system_root_dir.rb

```diff
@@ -110,6 +110,7 @@ def create_child(name, file_contents = nil)
               child_paths[name].each do |path|
                 begin
                   Dir.mkdir(path)
+                  File.chmod(0700, path) if name == "nodes"
```

Comment:
If we are going to do it for everything, I believe `mkdir` takes a second argument where you can set the mode on creation.